File: backend/lib/L1_domain/entities/auth/workspace.py
#  Copyright (c) 2022. Alexandr Moroz

from dataclasses import dataclass

from ..base_entity import Titleable
from ..goals import Goal, Person, RemoteTracker, TaskPriority, TaskStatus


@dataclass
class Workspace(Titleable):
    goals: list[Goal] = None
    task_statuses: list[TaskStatus] = None
    task_priorities: list[TaskPriority] = None
    persons: list[Person] = None
    remote_trackers: list[RemoteTracker] = None

File: frontend/lib/L1_domain/entities/auth/workspace.dart
// Copyright (c) 2022. Alexandr Moroz

import '../base_entity.dart';
import '../goals/goal.dart';
import '../goals/person.dart';
import '../goals/remote_tracker.dart';
import '../goals/task_priority.dart';
import '../goals/task_status.dart';

class Workspace extends Titleable {
  Workspace({
    required int id,
    required String title,
    required this.goals,
    required this.taskStatuses,
    required this.taskPriorities,
    required this.persons,
    required this.remoteTrackers,
  }) : super(id: id, title: title);

  final List<Goal> goals;
  final List<TaskStatus> taskStatuses;
  final List<TaskPriority> taskPriorities;
  final List<Person> persons;
  final List<RemoteTracker> remoteTrackers;
}
